sql/expression_value.cc

```diff
@@ -2608,6 +2608,68 @@ getNestedColumn(const ColumnName & columnName, const VariableFilter & filter) co
     else return *val;
 }
 
+ExpressionValue
+ExpressionValue::
+coerceToEmbedding() const
+{
+    if (type_ == Type::EMBEDDING)
+        return *this;
+    else if (type_ != Type::STRUCTURED)
+        throw HttpReturnException(500, "Cannot coerce value to embedding");
+
+    //Start with finding the shape of the embedding
+    size_t numElements = 0;
+    DimsVector shape;
+    auto atomCount = [&] (const Path & columnName,
+                                   const Path & prefix,
+                                   const CellValue & val,
+                                   Date ts)
+                    {
+                        Path fullColumnName = prefix + columnName;
+                        if (shape.size() < fullColumnName.size())
+                            shape.resize(fullColumnName.size());
+
+                        for (size_t i = 0; i < fullColumnName.size(); ++i) {
+                            shape[i] = std::max(shape[i], (size_t)(fullColumnName[i].toIndex()+1));
```

Comment:
This function should fail if the rows don't have a consistent size, as it's not an embedding in that case.